simulator/src/main/java/com/hazelcast/simulator/worker/WorkerMessageProcessor.java

```diff
@@ -92,22 +91,18 @@ private boolean isMaster() {
                 public Boolean call() throws Exception {
                     return isOldestMember();
                 }
-            }, 10, TimeUnit.SECONDS).get(TIMEOUT, TimeUnit.SECONDS);
-        } catch (InterruptedException e) {
-            throw new IllegalStateException(e);
-        } catch (ExecutionException e) {
-            throw new IllegalStateException(e);
-        } catch (TimeoutException e) {
+            }, delaySeconds, TimeUnit.SECONDS).get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        } catch (Exception e) {
             throw new IllegalStateException(e);
         }
     }
 
-    private boolean isOldestMember() {
+    boolean isOldestMember() {
```

Comment:
Hmm, could do that, but that class can't go into the utils module, since that one has no Hazelcast dependency (and I really try to limit that to the modules where it is necessary). So the util class would have to go into the simulator module.